src/bmpcodec.c

```diff
@@ -784,8 +796,29 @@ gdip_read_BITMAPINFOHEADER (void *pointer, ImageSource source, BITMAPV5HEADER *b
 	bmi->bV5ClrImportant = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
 
 	/* We've finished reading the BITMAPINFOHEADER but later versions are larger. */
-	if (bmi->bV5Size == sizeof (BITMAPINFOHEADER))
+	if (bmi->bV5Size == sizeof (BITMAPINFOHEADER)) {
+		if (bmi->bV5Compression == BI_BITFIELDS) {
+			dw = 0;
+			size_read = gdip_read_bmp_data (pointer, data_read, size, source);
+			if (size_read < size)
+				return OutOfMemory;
+			bmi->bV5RedMask = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
+
+			dw = 0;
+			size_read = gdip_read_bmp_data (pointer, data_read, size, source);
+			if (size_read < size)
+				return OutOfMemory;
+			bmi->bV5GreenMask = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
+
+			dw = 0;
+			size_read = gdip_read_bmp_data (pointer, data_read, size, source);
+			if (size_read < size)
+				return OutOfMemory;
+			bmi->bV5BlueMask = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
+		}
+		
```

Comment:
Any reason why you don't fall through to the code below? Also, now the implicit BI_RGB masks from previous change are missing...

Reply:
For the record I don't mind the code duplication if it makes the code more readable. It's matter of personal opinion and I find your version little more readable even though it's a bit verbose... but the BI_RGB part has to be fixed in either case.